frontend/ui/widget/opdsbrowser.lua

```diff
@@ -645,7 +660,55 @@ function OPDSBrowser:showDownloads(item)
     UIManager:show(self.download_dialog)
 end
 
+function OPDSBrowser:browse(browse_url, username, password)
+  logger.info("launch url", browse_url)
+  table.insert(self.paths, {
+                 url = browse_url,
+                 username = username,
+                 password = password,
+  })
+  if not self:updateCatalog(browse_url, username, password) then
+    table.remove(self.paths)
```

Comment:
Could you please observe the style guide? Four spaces indentation. :-) (That applies to everything in this function; I'm just attaching it to this line more or less at random.)

https://github.com/koreader/koreader-base/wiki/Coding-style